### DISCUSSION TITLE
Chapter granularity instead of the file granularity

### DISCUSSION THREAD

**Discussion by jixunx** [brianpetro/obsidian-smart-connections#573] [General]
How can the content associated through smart connection be at the chapter granularity instead of the file granularity? Currently, it can only be used at the file granularity

**Comment by brianpetro** [comment]
How do you organize your chapters?

Do you use headings?

If yes, you can turn on the "block-level" embeddings and that should effectively make embeddings per chapter.

🌴

**Reply by jixunx** [reply]
"Am I supposed to select the 'blocks embedding model' here? But I'm using OpenAI's API, so the embedding model should be invalid, right?"

**Reply by brianpetro** [reply]
@jixunx I'm not sure I understand what you're asking.

Regardless of the embedding model, if the note-level embeddings don't provide sufficiently granular output, then you should also use a block-level embedding model.

🌴

**Reply by jixunx** [reply]
How to use a block-level embedding model? I can't find out the method.

**Reply by jixunx** [reply]
Any one could answer my question?

**Reply by RealRavens** [reply]
Change the drop down to the desired embedding model where the arrow is pointing in the settings. 

[![Image from Gyazo](https://i.gyazo.com/0a8eecec79a88166e6969161d4c2aa42.png)](https://gyazo.com/0a8eecec79a88166e6969161d4c2aa42)

Some examples of "blocks" in obsidian are headings. Technically certain list elements and paragraphs have blocks but I'm not sure 100% if that's accounted for or implemented in smart notes. (could only imagine the size of those embedding files on large vaults).

**Reply by jixunx** [reply]
But this is local model. I am using chatGPT API model. So local model is invalid. Is that right?
![image](https://github.com/brianpetro/obsidian-smart-connections/assets/25680280/f372e54a-8f14-4659-85b1-bec13e349ea4)

**Reply by RealRavens** [reply]
I'm answering your question concerning the use of block embedding models. Just change none to any of the models and it will begin to embed your note "blocks" throughout your vault. If you're using chatgpt aka OpenAI, you select one of the ones in this picture:

[![Image from Gyazo](https://i.gyazo.com/00e4e18d3c5a32f962c6bfbd83787dd4.png)](https://gyazo.com/00e4e18d3c5a32f962c6bfbd83787dd4)

So in my pervious picture I was pointing at the drop down list for block embedding models. Select the one you like, and in this case will be one of the OpenAI models. Ofc including your API key where required. Hope that helps?

**Comment by jixunx** [comment]
OK， It is success.  Thanks to RealRaven !

**Reply by RealRavens** [reply]
You're very welcome. Take care ~
